src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/Buffering/HttpRequestBufferProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if NET9_0_OR_GREATER
+using System.Collections.Concurrent;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Diagnostics.Logging;
+
+internal sealed class HttpRequestBufferProvider : ILoggingBufferProvider
+{
+    private readonly GlobalBufferProvider _globalBufferProvider;
```

Comment:
This works because `Microsoft.Extensions.Telemetry` has `InternalsVisibleTo` to `Microsoft.AspNetCore.Diagnostics.Middleware` and I am just taking advantage of that. Are you suggesting to avoid that? I mean, that would force us to make some types `public` just to make them visible for `Microsoft.AspNetCore.Diagnostics.Middleware`, while we probably don't have an intention to make them visible for everyone

Reply:
> Also I notice HttpRequestBufferOptions has a GlobalCapacity property. I'm not clear if someone wanted to set the capacity are they expected to set that property or the GlobalBufferOptions.Capacity property?

Removed `GlobalCapacity` from `HttpRequestBufferOptions`, keeping it only in `GlobalBufferOptions`. The intention is that we provide reasonable defaults out of the box, and if someone wants to change that, they would configure Options as usual.